src/Symfony/Component/Message/Handler/MessageHandlerCollection.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message\Handler;
+
+/**
+ * Represents a collection of message handlers.
+ *
+ * @author Samuel Roze <samuel.roze@gmail.com>
+ */
+class MessageHandlerCollection
```

Comment:
`ChainHandler`?

Reply:
Naming was already [discussed here](https://github.com/symfony/symfony/pull/24411#discussion_r143429868). I don't believe there is more value in changing again this name tbf.

Reply:
well.. it's not a _collection_, it's a _handler_. Hence i think it needs a different suffix, i.e. `CollectionHandler` would make more senso also IMHO, but i think `Chain` is more common for this pattern implem.

Reply:
`HandlerCollection` then? I mean  a _handler_ already implies a _message handler_ no?

Reply:
Alright, we'll get a vote:
- 😄 : We don't have to change anything
- 👍  : `CollectionHandler`
- 🎉 : `HandlerCollection`
- 😕  : `ChainHandler`

Reply:
We don't need a vote 😄  Just think which single word would you pick to best describe this thing: "handler", "collection", "chain", etc.  (@ro0NL says that this is a "handler"). Then, pick the other word to complete the meaning ("collection" or "chain" in this case). So ... the best name is probably `CollectionHandler` or `ChainHandler`.

Reply:
IHMO `Chain` is the standard name for such things.

Reply:
Changed to `ChainHandler` :)